src/HttpRequest.js

```diff
@@ -18,10 +23,16 @@ const HttpRequest = {
 	},
 
 	build(data = {}) {
-		data.url != null ? this.url = data.url : null;
-		data.headers != null ? this.headers = data.headers : null;
-		data.body != null ? this.body = data.body : null;
-		data.json != null ? this.json = data.json : null;
+		const {url, headers, body, json, qs, resolveWithFullResponse, ...options} = data;
+
+		url != null     && (this.url = url);
+		headers != null && (this.headers = headers);
+		body != null    && (this.body = body);
+		json != null    && (this.json = json);
+		qs != null      && (this.qs = qs);
+		resolveWithFullResponse != null && (this.resolveWithFullResponse = resolveWithFullResponse);
+		this.options = options;
```

Comment:
Do we want a null check here? Won't this add empty `options` params onto the object if none are passed in?

Reply:
For the following assignment, other is an empty object, not undefined.

```js
const {a, b, c, ...other} = {};
console.log(other); // {}
```

Since `data` is defaulted to `{}`, there are no situations where `options` is null or undefined.

Reply:
If we want data to leave the existing fields as they were before calling build, that can be done but it would require a check like `Object.keys(options).length === 0 && (this.options = options);`